packages/overlay/src/vaadin-overlay-mixin.js

```diff
@@ -469,4 +474,14 @@ export const OverlayMixin = (superClass) =>
         }
       }
     }
+
+    /** @private */
+    _onMouseDown(event) {
+      const target = event.target;
+      const focusable = getClosestFocusable(target.focusElement || target);
```

Comment:
Checking for `focusElement` is needed for cases when an event target has its own overlay and needs to bring it to front (see `should not end up behind the dialog overlay on mousedown` IT test).

Reply:
Could / should we have a test case for this (`target.focusElement`) here also?

Reply:
Added a test.